### PR TITLE
Fix greenspline usage and add -f

### PR DESCRIPTION
**Pull request by PaulWessel**
Follows the recipe of #5341.  Noticed that -f was missing in usage.

### CODE REVIEW

File: doc/rst/source/greenspline.rst [hunk 1]
```diff
@@ -26,7 +26,7 @@ Synopsis
 [ |-S|\ **c\|t\|l\|r\|p\|q**\ [*pars*] ] [ |-T|\ *maskgrid* ]
```
Comment (meghanrjones) [inline doc/rst/source/greenspline.rst:26]: -G should be moved to required section for consistency

File: src/greenspline.c [hunk 2]
```diff
@@ -227,85 +227,103 @@ static void Free_Ctrl (struct GMT_CTRL *GMT, struct GREENSPLINE_CTRL *C) {	/* De
 static int usage (struct GMTAPI_CTRL *API, int level) {
 	const char *name = gmt_show_name_and_purpose (API, THIS_MODULE_LIB, THIS_MODULE_CLASSIC_NAME, THIS_MODULE_PURPOSE);
 	if (level == GMT_MODULE_PURPOSE) return (GMT_NOERROR);
-	GMT_Message (API, GMT_TIME_NONE, "usage: %s [<table>] -G<outfile> [-A<gradientfile>+f<format>] [-C[n]<val>[%%][+f<file>][+m|M]]\n", name);
-	GMT_Message (API, GMT_TIME_NONE, "\t[-D<information>] [-E[<misfitfile>] [-I<dx>[/<dy>[/<dz>]] [-L] [-N<nodefile>] [-Q<az>]\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t[-R<xmin>/<xmax[/<ymin>/<ymax>[/<zmin>/<zmax>]]] [-Sc|l|t|r|p|q[<pars>]] [-T<maskgrid>]\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t[%s] [-W[w]] [-Z<mode>] [%s] [%s] [%s]\n\t[%s] [%s]\n\t[%s] [%s]\n\t[%s] [%s] [%s]\n\t[%s]%s[%s] [%s]\n\n", GMT_V_OPT,
-		GMT_bi_OPT, GMT_d_OPT, GMT_e_OPT, GMT_g_OPT, GMT_h_OPT, GMT_i_OPT, GMT_o_OPT, GMT_q_OPT, GMT_r_OPT, GMT_s_OPT, GMT_w_OPT, GMT_x_OPT, GMT_colon_OPT, GMT_PAR_OPT);
+	GMT_Usage (API, 0, "usage: %s [<table>] -G<outfile> [-A<gradientfile>+f<format>] [-C[n]<val>[%%][+f<file>][+m|M]] "
+		"[-D<information>] [-E[<misfitfile>] [-I<dx>[/<dy>[/<dz>]]] [-L] [-N<nodefile>] [-Q<az>] "
```
Comment (meghanrjones) [inline src/greenspline.c:231]:
```suggestion
		"[-D<information>] [-E[<misfitfile>]] [-I<dx>[/<dy>[/<dz>]]] [-L] [-N<nodefile>] [-Q<az>] "
```

File: doc/rst/source/greenspline.rst [hunk 3]
```diff
@@ -26,7 +26,7 @@ Synopsis
...
 [ |SYN_OPT-V| ]
 [ |-W|\ [**w**]]
-[ |-Z|\ *mode* ]
+[ |-Z|\ *0-5* ]
 [ |SYN_OPT-b| ]
 [ |SYN_OPT-d| ]
 [ |SYN_OPT-e| ]
```
Comment (meghanrjones) [inline doc/rst/source/greenspline.rst:32]: -g, -i, -r and -s are missing

File: src/greenspline.c [hunk 4]
```diff
@@ -227,85 +227,103 @@ static void Free_Ctrl (struct GMT_CTRL *GMT, struct GREENSPLINE_CTRL *C) {	/* De
...
+		"A negative cutoff will stop execution after reporting the eigenvalues. "
+		"Use -Cn to select only the largest <val> eigenvalues [all]. "
+		"Use <val>%% to select a percentage of the eigenvalues instead "
+		"[Default uses Gauss-Jordan elimination to solve the linear system].");
+	GMT_Usage (API, 3, "+f Save the eigenvalues to <filename>.");
+	GMT_Usage (API, 3, "+m Valid for 2-D gridding only and will create a series of intermediate "
+		"grids for each eigenvalue holding the incremental result. Requires -G with a filename "
+		"template containing an integer C formatting specifier (e.g., %%d).");
+	GMT_Usage (API, 3, "+M Same, but for cumulative results.");
 	gmt_grdcube_info_syntax (API->GMT, 'D');
```
Comment (meghanrjones) [inline src/greenspline.c:271]: -D+n is missing

File: src/greenspline.c [hunk 5]
```diff
@@ -227,85 +227,103 @@ static void Free_Ctrl (struct GMT_CTRL *GMT, struct GREENSPLINE_CTRL *C) {	/* De
...
-	GMT_Message (API, GMT_TIME_NONE, "\t   Append filename to save all data with two extra columns for model and misfit.\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t-I Specify a regular set of output locations.  Give equidistant increment for each dimension.\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t   Requires -R for specifying the output domain.\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t-L Leave trend alone.  Do not remove least squares plane from data before spline fit.\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t   Only applies to -D0-2.  [Default removes linear trend, fits residuals, and restores trend].\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t-N ASCII file with desired output locations.\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t   The resulting ASCII coordinates and interpolation are written to file given in -G\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t   or stdout if no file specified (see -bo for binary output).\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t-Q Calculate the directional derivative in the <az> direction and return it instead of surface elevation.\n");
+	GMT_Usage (API, 1, "\n-E[<misfitfile>");
```
Comment (meghanrjones) [inline src/greenspline.c:272]:
```suggestion
	GMT_Usage (API, 1, "\n-E[<misfitfile>]");
```

**Review by meghanrjones — APPROVED** [review]

File: doc/rst/source/greenspline.rst (hunk 3 above)
Comment (PaulWessel) [reply doc/rst/source/greenspline.rst:32]: Done

File: src/greenspline.c (hunk 4 above)
Comment (PaulWessel) [reply src/greenspline.c:271]: Thanks, added

File: doc/rst/source/greenspline.rst (hunk 1 above)
Comment (PaulWessel) [reply doc/rst/source/greenspline.rst:26]: OK done, and fixed -D via new SYN for those modules using it.

**Pull request merged**